### PR TITLE
Use 6 decimals while getting unit price

### LINKED ISSUE #31025: Used decimals while creating order detail are not the same and there is an extra quantity discount
### Prerequisites

- [X] I understand and accept the project's [code of conduct](https://github.com/PrestaShop/PrestaShop/blob/develop/CODE_OF_CONDUCT.md).
- [X] I have already [searched in existing issues](https://github.com/PrestaShop/PrestaShop/issues?q=is%3Aissue+label%3ABug) and found no previous report of this bug.

### Describe the bug and add attachments

When you order a product, you could have the colomn `product_quantity_discount` having the `0.000001` value inside it even if your product have no discount on it.

### Expected behavior

Having the value 0 inside this colum.

### Steps to reproduce

Add a product with theses settings : 

- Tax rules : 21%
- Base price (without taxes) : 119€

Set a specific price on it :
- Set the price to 98.347107 and uncheck the "Keep the base price" setting.

Order this product and check the order_details row added.

### PrestaShop version(s) where the bug happened

1.7.8.x, 8.0.1

### PHP version(s) where the bug happened

_No response_

### If your bug is related to a module, specify its name and its version

_No response_

### PR DESCRIPTION
**Pull request by PrestaEdit**
| Questions         | Answers
| ----------------- | -------------------------------------------------------
| Branch?           | develop
| Description?      | Use 6 decimals to avoid false calculation, as seen on the linked issue.
| Type?             | bug fix
| Category?         | CO
| BC breaks?        | no
| Deprecations?     | no
| Fixed ticket?     | Fixes #31025
| Sponsor company   | @Wepika

### CODE REVIEW

File: classes/order/OrderDetail.php [hunk 1]
```diff
@@ -706,7 +706,7 @@ protected function setDetailProductPrice(Order $order, Cart $cart, $product)
             (int) $product['id_product'],
             true,
             ($product['id_product_attribute'] ? (int) ($product['id_product_attribute']) : null),
-            2,
+            6,
```
Comment (matks) [inline classes/order/OrderDetail.php:709]: I might be wrong, but should'nt we use `Context::getContext()->getComputingPrecision()` ?

**Comment by kpodemski** [comment]
The fix seems legit, but QA didn't reproduce the error.

@PrestaEdit could you verify the steps you provided?

**Comment by PrestaEdit** [comment]
QA have validated the use case.

**Review by kpodemski — APPROVED** [review]

File: classes/order/OrderDetail.php (hunk 1 above)
Comment (PululuK) [reply classes/order/OrderDetail.php:709]: ping @PrestaEdit

Comment (PrestaEdit) [reply classes/order/OrderDetail.php:709]: As the rest of the code use it as it, I will say : no ?

**Comment by PrestaEdit** [comment]
Hi,

There is no milestone on this one. Could we add one, please ?

**Comment by PrestaEdit** [comment]
Hi,

Does this need something to be reviewable ?

**Review by ghost — APPROVED** [review]

**Review by zuk3975 — APPROVED** [review]

**Review by MhiriFaten — COMMENTED** [review]
Hello @PrestaEdit ,

The issue is well fixed and LGTM :heavy_check_mark: 


But the automated tests are failed :warning: 
https://github.com/MhiriFaten/testing_pr/actions/runs/4982037907

Ping @PrestaShop/qa-automation could you check it please?

Thank you!

**Comment by nesrineabdmouleh** [comment]
Hello @MhiriFaten,
Thank you for your tests :) the errors are not related to the PR, it's a regression in the develop branch.

**Comment by nicosomb** [comment]
Thank you @PrestaEdit !

**Pull request merged**